operator/internal/manifests/rules_config.go

```diff
@@ -10,39 +10,41 @@ import (
 
 // RulesConfigMap returns a ConfigMap resource that contains
 // all loki alerting and recording rules as YAML data.
-func RulesConfigMap(opts *Options) (*corev1.ConfigMap, error) {
-	data := make(map[string]string)
+// If the size of the data is more than 1MB, the ConfigMap will
+// be split into multiple shards, and this function will return
+// the list of shards
+func RulesConfigMapShards(opts *Options) ([]*corev1.ConfigMap, error) {
+	l := commonLabels(opts.Name)
+
+	template := newConfigMapTemplate(opts, l)
+
+	shardedConfigMap := NewShardedConfigMap(template, RulesConfigMapName(opts.Namespace))
```

Comment:
nit. Rename `shardedConfigMap` to `shardedCM`. Short names to remove duplication with `NewShardedConfigMap`